### PR TITLE
feat: Beyond Light support

### PR DESCRIPTION
**Pull request by nblockbuster**

### CODE REVIEW

File: Tiger/Schema/Activity/Activity.cs [hunk 1]
```diff
@@ -88,3 +88,46 @@ public IEnumerable<Bubble> EnumerateBubbles()
...
+    {
+        public FileHash FileHash => Hash;
+
+        public Activity(FileHash hash) : base(hash)
+        {
+        }
+
+        public IEnumerable<Bubble> EnumerateBubbles()
+        {
+            // TODO: i dont like this solution
```
Comment (MontagueM) [inline Tiger/Schema/Activity/Activity.cs:103]: solutions may not be best but if they work im happy

File: Tiger/Schema/Static/StaticMeshData.cs [hunk 2]
```diff
@@ -103,10 +103,107 @@ private List<StaticPart> GenerateParts(Dictionary<int, SStaticMeshPart> staticPa
     }
 }
 
+//TODO: collapse this into witch queen version?
```
Comment (MontagueM) [inline Tiger/Schema/Static/StaticMeshData.cs:106]: can you do this?

File: Tiger/Schema/Static/StaticMeshStructs.cs [hunk 3]
```diff
@@ -60,14 +60,15 @@ public struct SStaticMeshData_SK
...
 [SchemaStruct(TigerStrategy.DESTINY2_WITCHQUEEN_6307, "306D8080", 0x60)]
-public struct SStaticMeshData_WQ
+public struct SStaticMeshData_BL
 {
     public long FileSize;
     public DynamicArray<SStaticMeshMaterialAssignment_WQ> MaterialAssignments;
     public DynamicArray<SStaticMeshPart> Parts;
     public DynamicArray<SStaticMeshBuffers> Meshes;
-    [SchemaField(0x40)]
+    [SchemaField(0x50)]
```
Comment (MontagueM) [inline Tiger/Schema/Static/StaticMeshStructs.cs:71]: is this changed bc WQ was wrong, or because BL is different?

File: Tiger/Schema/Static/StaticPart.cs [hunk 4]
```diff
@@ -109,9 +109,9 @@ public void GetDecalData(SStaticMeshDecal mesh, SStaticMesh container)
 
     private void TransformData(SStaticMesh container)
     {
-        if (Strategy.CurrentStrategy >= TigerStrategy.DESTINY2_WITCHQUEEN_6307)
+        if (Strategy.CurrentStrategy >= TigerStrategy.DESTINY2_BEYONDLIGHT_3402)
         {
-            var t = (container.StaticData as Tiger.Schema.Static.DESTINY2_WITCHQUEEN_6307.StaticMeshData).TagData;
+            var t = (container.StaticData as dynamic).TagData;
```
Comment (MontagueM) [inline Tiger/Schema/Static/StaticPart.cs:114]: why this change?

File: Tiger/Schema/Entity/EntityStructs.cs [hunk 5]
```diff
@@ -118,21 +118,25 @@ public struct D2Class_6E908080
...
-[SchemaStruct(TigerStrategy.DESTINY2_WITCHQUEEN_6307, "8F6D8080", 0x450)]
+[SchemaStruct(TigerStrategy.DESTINY2_BEYONDLIGHT_3402, "8F6D8080", 0x450)]
 public struct D2Class_8F6D8080
 {
     [SchemaField(0x1DC, TigerStrategy.DESTINY2_SHADOWKEEP_2601)]
-    [SchemaField(0x224, TigerStrategy.DESTINY2_WITCHQUEEN_6307)]
+    [SchemaField(0x224, TigerStrategy.DESTINY2_BEYONDLIGHT_3402)]
     public EntityModel Model;
-    [SchemaField(0x310, TigerStrategy.DESTINY2_WITCHQUEEN_6307)]  // todo shadowkeep
+    [SchemaField(0x310, TigerStrategy.DESTINY2_WITCHQUEEN_6307)]  // todo shadowkeep, beyond light
```
Comment (MontagueM) [inline Tiger/Schema/Entity/EntityStructs.cs:127]: is this actually todo in future? if so id probably disable the API for BL (the code is in the mainmenu somewhere)

File: Tiger/Schema/Entity/Entity.cs [hunk 6]
```diff
@@ -38,7 +38,8 @@ public void Load()
         Debug.Assert(_tag.FileSize != 0);
         foreach (var resource in _tag.EntityResources.Select(GetReader(), r => r.Resource))
         {
-            switch (resource.TagData.Unk10.GetValue(resource.GetReader()))
+            var res2 = resource.TagData.Unk10.GetValue(resource.GetReader());
+            switch (res2)
```
Comment (MontagueM) [inline Tiger/Schema/Entity/Entity.cs:42]: for debugging purposes? revert ideally

File: Tiger/Schema/Audio/Dialogue.cs [hunk 7]
```diff
@@ -53,8 +57,10 @@ public Dialogue(FileHash hash) : base(hash)
     private List<dynamic?> Collapse2D97(D2Class_2D978080 entry)
     {
         List<dynamic?> sounds = new();
-        foreach (dynamic? e in entry.Unk20.Select(u => u.Unk20.GetValue(GetReader())))
+        foreach (var u in entry.Unk20)
```
Comment (MontagueM) [inline Tiger/Schema/Audio/Dialogue.cs:60]: also debugging?

File: Charm/MainViewModel.cs [hunk 8]
```diff
@@ -310,7 +310,7 @@ public void SetChildren(List<DisplayPart> parts)
     private Vector3 ConsiderQuatToEulerConvert(Vector4 v4N)
     {
         // shadowkeep and below don't have quaternion normals
-        if (Strategy.CurrentStrategy < TigerStrategy.DESTINY2_WITCHQUEEN_6307)
+        if (Strategy.CurrentStrategy < TigerStrategy.DESTINY2_BEYONDLIGHT_3402)
```
Comment (MontagueM) [inline Charm/MainViewModel.cs:313]: could change to >= 2999

File: Tiger/Schema/Entity/EntityStructs.cs (hunk 5 above)
Comment (MontagueM) [reply Tiger/Schema/Entity/EntityStructs.cs:127]: nvm saw you did this

**Comment by MontagueM** [comment]
other than my comments, as long as 2999 and 6307 have been tested and work particularly with activities, awesome work and very cool change

File: Tiger/Schema/Static/StaticMeshStructs.cs (hunk 3 above)
Comment (nblockbuster) [reply Tiger/Schema/Static/StaticMeshStructs.cs:71]: BL is different, forgot to add the strategy to the attribute, will be fixed in the next commit

File: Tiger/Schema/Static/StaticPart.cs (hunk 4 above)
Comment (nblockbuster) [reply Tiger/Schema/Static/StaticPart.cs:114]: it was crashing on witch queen versions and above, but it won't be dynamic anymore with merging BL and WQ's StaticMeshData

File: Tiger/Schema/Activity/Activity.cs (hunk 1 above)
Comment (nblockbuster) [reply Tiger/Schema/Activity/Activity.cs]: made it slightly cleaner but true enough, it does work

File: Tiger/Schema/Static/StaticMeshData.cs (hunk 2 above)
Comment (nblockbuster) [reply Tiger/Schema/Static/StaticMeshData.cs]: done in 3b61be6cc1e9564f5efe94f85e0502198fb92dc4

**Comment by nblockbuster** [comment]
tested everything with 2999, 3402, 6307, latest and everything seems to be working

**Review by MontagueM — APPROVED** [review]
